### PR TITLE
Dump SQL for features

### PR DESCRIPTION
**Pull request by pierallard**
Currently, each step `Given a xx catalog`
- loads the fixtures jobs
- loads the fixtures
- removes the fixtures jobs
For example, the bigger catalog (catalog_modeling) takes ~50s.

This PR dumps the SQL after a catalog initialization, in a .sql file. In the next catalog initialization, if this .sql already exists, it only reloads the SQL file.
For example, catalog_modeling is loaded in ~7s.

Previous timings : 34m08, 34m14
https://circleci.com/gh/akeneo/pim-community-dev/4526#build-timing
https://circleci.com/gh/akeneo/pim-community-dev/4991#build-timing

New timing: 28m57
https://circleci.com/gh/akeneo/pim-community-dev/5106#build-timing

- [x] To check before merge: Why we didn't put the .sql in the ~/var/cache folder ? Same question for the Integration tests .sql files.

### CODE REVIEW

File: tests/legacy/features/Context/Loader/FixturesLoader.php [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+     * @param string[] $configurationFiles
+     */
+    private function loadCatalogOrRestoreDatabase(array $configurationFiles)
+    {
+        $fixturesHash = $this->getHashForFiles($configurationFiles);
+        $dumpFile = sys_get_temp_dir() . self::CACHE_DIR . $fixturesHash . '.sql';
+
+        if (file_exists($dumpFile)) {
+            $this->restoreDatabase($dumpFile);
+            $this->getProductModelIndexer()->indexAll($this->getProductModelRepository()->findAll());
```
Comment (skeleton) [inline tests/legacy/features/Context/Loader/FixturesLoader.php]: just to be sure, you don't need to clear the index before adding product model to ES?

Comment (ahocquard) [reply tests/legacy/features/Context/Loader/FixturesLoader.php]: `$this->resetElasticsearchIndex();` do the job

Comment (skeleton) [reply tests/legacy/features/Context/Loader/FixturesLoader.php]: ok, I did not see it, thank

**Comment by ahocquard** [comment]
Done in another PR :)

**Pull request closed without merging**